src/frequenz/channels/experimental/_pipe.py

```diff
@@ -25,17 +25,34 @@ class Pipe(typing.Generic[ChannelMessageT]):
 
     Example:
         ```python
-        from frequenz.channels import Broadcast, Pipe
-
-        channel1: Broadcast[int] = Broadcast(name="channel1")
-        channel2: Broadcast[int] = Broadcast(name="channel2")
-
-        receiver_chan1 = channel1.new_receiver()
-        sender_chan2 = channel2.new_sender()
-
-        async with Pipe(channel2.new_receiver(), channel1.new_sender()):
-            await sender_chan2.send(10)
-            assert await receiver_chan1.receive() == 10
+        from contextlib import asynccontextmanager
+        from typing import AsyncIterator
+        from frequenz.channels import Broadcast, Pipe, Receiver
+
+        @asynccontextmanager
+        async def create_forwarding_pipe(
+            receiver: Receiver[int],
+        ) -> AsyncIterator[Broadcast[int]]:
+            forwarding_channel: Broadcast[int] = Broadcast(name="forwarded channel")
+            forwarding_sender = forwarding_channel.new_sender()
+            try:
+                async with Pipe(receiver, forwarding_sender):
+                    yield forwarding_channel
+            finally:
+                receiver.close()
+                await forwarding_channel.close()
+
+        source_channel: Broadcast[int] = Broadcast(name="source channel")
+        source_sender = source_channel.new_sender()
+
+        # Create a pipe in a context manager to ensure the
+        # resources are cleaned up when the context is exited.
+        async with create_forwarding_pipe(
+            source_channel.new_receiver()
+        ) as forwarding_channel:
+            receiver = forwarding_channel.new_receiver()
```

Comment:
I guess here you don't close the receiver because you are closing the channel? Maybe this needs a note because it might not be obvious? I also wonder if, at least for this example, it wouldn't make more sense to return a new receiver instead of the whole channel, and explicitly close the receiver in the async context manager, to make it more clear/explicit.